### PR TITLE
sara-abdulbasit

### PR DESCRIPTION
**Pull request by abdulbasit-dev**
complete task please accepte

### CODE REVIEW

File: forms-html/sara_abdulbasit/index.html [hunk 1]
```diff
@@ -0,0 +1,70 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <title>Form</title>
+</head>
+
+<body>
+  <li>make the email input required</li>
```
Comment (osamaakb) [inline forms-html/sara_abdulbasit/index.html:10]: why using li tag inside body directly?

File: forms-html/sara_abdulbasit/index.html [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  <br>
+  <br>
+  <form>
+    <label for="email">ENTER YOUR EMAIL</label> <br>
+    <input type="email" id="email" name="email" value="osama@re-coded.com"><br>
```
Comment (osamaakb) [inline forms-html/sara_abdulbasit/index.html:15]: the email should be required, when the user clicks the submit button it should ask him to fill it if it's empty

File: forms-html/sara_abdulbasit/index.html [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    <label>Username</label> <br>
+    <input type="text" name="username" value="osama"><br>
+    <label for="password">password:</label><br>
+    <input type="password" name="password" value="*****">
+    <br>
+    <label for="number">Quantity(between 1 and 5)</label>
+    <input type="number" id="number" min="1" max="5" value="3"><br>
+    
+
+    <label for="range" >volume(between 0 and 50):</label>
```
Comment (osamaakb) [inline forms-html/sara_abdulbasit/index.html:25]: should have max, min value and id

File: forms-html/sara_abdulbasit/index.html [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    <input type="checkbox" name="chexkbox" id="boat" value=" I have a boat">
+    <label for="boat"> i have a boat</label>
+
+    <p>button</p>
+    <input type="submit" value="click me"><br>
+    <label for="color">Select your favorate color</label>
+    <input type="color" name="color" id="color"><br>
+    
+
+ <label for="" >Enter a date before 1980-01-01:</label>
```
Comment (osamaakb) [inline forms-html/sara_abdulbasit/index.html:52]: shouldn't accept any date after 1980-1-01

File: forms-html/sara_abdulbasit/index.html [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+  <li>make the email input required</li>
+  <br>
+  <br>
+  <form>
+    <label for="email">ENTER YOUR EMAIL</label> <br>
+    <input type="email" id="email" name="email" value="osama@re-coded.com"><br>
+    <label>Username</label> <br>
+    <input type="text" name="username" value="osama"><br>
+    <label for="password">password:</label><br>
+    <input type="password" name="password" value="*****">
```
Comment (osamaakb) [inline forms-html/sara_abdulbasit/index.html:19]: missing id here 😅

**Review by osamaakb — COMMENTED** [review]
you guys did an excellent job here, left you some notes.
Also please remove the .vscode folder.

**Comment by abdulbasit-dev** [comment]
thank you

**Pull request closed without merging**